From 381b9a18ce0ee61d24c32e8cb659bc10a1efd651 Mon Sep 17 00:00:00 2001
From: Yang Yingliang <yangyingliang@huawei.com>
Date: Tue, 12 Apr 2022 16:15:53 +0300
Subject: [PATCH 199/241] ath11k: fix missing unlock on error in
 ath11k_wow_op_resume()

Add the missing unlock before return from function ath11k_wow_op_resume()
in the error handling case.

Fixes: 90bf5c8d0f7e ("ath11k: purge rx pktlog when entering WoW")
Reported-by: Hulk Robot <hulkci@huawei.com>
Signed-off-by: Yang Yingliang <yangyingliang@huawei.com>
Signed-off-by: Kalle Valo <quic_kvalo@quicinc.com>
Link: https://lore.kernel.org/r/20220408030912.3087293-1-yangyingliang@huawei.com
---
 drivers/net/wireless/ath/ath11k/wow.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

--- a/drivers/net/wireless/ath/ath11k/wow.c
+++ b/drivers/net/wireless/ath/ath11k/wow.c
@@ -758,7 +758,7 @@ int ath11k_wow_op_resume(struct ieee8021
 	ret = ath11k_dp_rx_pktlog_start(ar->ab);
 	if (ret) {
 		ath11k_warn(ar->ab, "failed to start rx pktlog from wow: %d\n", ret);
-		return ret;
+		goto exit;
 	}
 
 	ret = ath11k_wow_wakeup(ar->ab);
